src/mongocxx/search_index_model.hpp

```diff
@@ -32,7 +32,7 @@ class MONGOCXX_API search_index_model {
     ///
     /// Move assigns a search_index_model.
     ///
-    search_index_model& operator=(search_index_model&&) noexcept;
```

Comment:
Marking a function as `noexcept` requires that all members of the class must also have `noexcept` on the corresponding function. While the `_definition` field is a `bsoncxx::document::value` which has its move assign operator as `noexcept`, the ` bsoncxx::stdx::optional<std::string> _name` field does not have the same tag on its move assign operator. Instead of making changes to `stdx::optional` which could affect the entire codebase, it's easier to just remove the `noexcept` from the move assign operator.